src/zope/testing/renormalizing.py

```diff
@@ -70,3 +94,31 @@ def output_difference(self, example, got, optionflags):
 
 RENormalizing = OutputChecker
 
+
+def maybe_a_traceback(string):
+    if not string:
+        return None
+
+    lines = string.splitlines()
+    last = lines[-1]
+    words = last.split(' ')
+    first = words[0]
+    if not first.endswith(':'):
+        return None
+
+    return lines, last, words, first
+
+
+def strip_dottedname_from_traceback(string):
+    # We might want to confirm more strictly were dealing with a traceback.
```

Comment:
I'd prefer that.  See [doctest.py](https://github.com/python/cpython/blob/36ebdb1812fd7676f3a25f872158cc4a9503881a/Lib/doctest.py#L604-L615) for the *sigh* private regexp.  I think a simpler version that checks just the first line would suffice.